tests/DCPS/Compiler/xcdr/xcdr.cpp

```diff
@@ -2188,198 +2184,225 @@ void set_values<ComplexKeyedStruct, KeyOnly<ComplexKeyedStruct> >(ComplexKeyedSt
 }
 
 template<typename TypeA, typename TypeB>
-void key_only_expect_values_equal_base_keys(const TypeA& a, const TypeB& b)
-{
-  EXPECT_EQ(a.long_value, b.long_value);
-  expect_arrays_are_equal(a.long_array_value, b.long_array_value);
-  /* TODO(iguessthislldo): See IDL Def
-  EXPECT_EQ(a.long_seq_value, b.long_seq_value);
-  */
-  EXPECT_STREQ(a.string_value, b.string_value);
-}
-
-template<typename TypeA, typename TypeB>
-void key_only_expect_values_equal_base(const TypeA& a, const TypeB& b)
+void key_only_tests_struct_expect_values_equal(
+  const TypeA& a, const TypeB& b,
+  FieldFilter field_filter, bool keyed)
 {
-  key_only_expect_values_equal_base_keys(a, b);
-  EXPECT_EQ(a.extra_value, b.extra_value);
+  if (!(field_filter == FieldFilter_KeyOnly && !keyed)) {
+    EXPECT_EQ(a.long_value, b.long_value);
+    expect_arrays_are_equal(a.long_array_value, b.long_array_value);
+    /* TODO(iguessthislldo): See IDL Def
+    EXPECT_EQ(a.long_seq_value, b.long_seq_value);
+    */
```

Comment:
There is still a TODO here, is that expected?

Reply:
Yes, it says what to look at, which points to the Jira ticket.